### PR TITLE
Update CMakeLists.txt and V8Inspector for React Native 0.73 compatibility

### PR DESCRIPTION
**Pull request by Serck96**

### CODE REVIEW

File: android/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,12 @@ cmake_minimum_required(VERSION 3.13)
 project(react-native-v8)
 
 set(CMAKE_VERBOSE_MAKEFILE ON)
-set(CMAKE_CXX_STANDARD 17)
+if(${REACT_NATIVE_TARGET_VERSION} GREATER_EQUAL 73)
+    add_compile_definitions(REACT_NATIVE_0_73)
```
Comment (Kudo) [inline android/CMakeLists.txt:14]:
```suggestion
```

we could reuse the `REACT_NATIVE_TARGET_VERSION`

File: src/v8runtime/V8Inspector.cpp [hunk 2]
```diff
@@ -15,7 +15,11 @@
 #include "folly/json.h"
 #include "v8.h"
 
+#ifdef REACT_NATIVE_0_73
```
Comment (Kudo) [inline src/v8runtime/V8Inspector.cpp:18]:
```suggestion
#if REACT_NATIVE_TARGET_VERSION >= 73
```

we could reuse the `REACT_NATIVE_TARGET_VERSION`

File: src/v8runtime/V8Inspector.h [hunk 3]
```diff
@@ -9,8 +9,14 @@
...
 #include <condition_variable>
-#include "jsinspector/InspectorInterfaces.h"
 #include "v8-inspector.h"
+#ifdef REACT_NATIVE_0_73
+#include "jsinspector-modern/InspectorInterfaces.h"
+using IRemoteConnection = facebook::react::jsinspector_modern::IRemoteConnection;
+#else
+#include "jsinspector/InspectorInterfaces.h"
+using IRemoteConnection = facebook::react::IRemoteConnection;
+#endif
```
Comment (Kudo) [inline src/v8runtime/V8Inspector.h:19]: maybe we can use class forward declaration and move these "includes" into V8Inspector.cpp

```cpp
namespace facebook {
namespace react {

class IRemoteConnection;

} // namespace react
} // namespace facebook
```

**Review by Kudo — COMMENTED** [review]
great pr! thanks for updating this. leaving some comments and after that it should be good to merge

**Comment by verluci** [comment]
Could you update the documentation to reflect the changes needed in MainApplication.kt (previously MainApplication.java)?

**Comment by Serck96** [comment]
Hi @verluci ,

I just wanted to let you know that I've updated the documentation with the changes you requested for MainApplication.kt. The updated configuration and instructions are now available in the repository's documentation.

Feel free to check it out and let me know if it works for you or if there are any further adjustments needed.

File: src/v8runtime/V8Inspector.h [hunk 4]
```diff
@@ -9,8 +9,14 @@
 
 #include <cxxreact/MessageQueueThread.h>
 #include <condition_variable>
-#include "jsinspector/InspectorInterfaces.h"
 #include "v8-inspector.h"
+#if REACT_NATIVE_TARGET_VERSION >= 73
+#include "jsinspector-modern/InspectorInterfaces.h"
+namespace react = facebook::react::jsinspector_modern;
```
Comment (Kudo) [inline src/v8runtime/V8Inspector.h:15]: this feels not quite safe to alias jsinspector_modern as react namespace.
how about just use jsinspector?

```diff
diff --git a/node_modules/react-native-v8/src/v8runtime/V8Inspector.cpp b/node_modules/react-native-v8/src/v8runtime/V8Inspector.cpp
index 2976039..7a1fb13 100644
--- a/node_modules/react-native-v8/src/v8runtime/V8Inspector.cpp
+++ b/node_modules/react-native-v8/src/v8runtime/V8Inspector.cpp
@@ -15,8 +15,6 @@
 #include "folly/json.h"
 #include "v8.h"
 
-namespace react = facebook::react;
-
 namespace rnv8 {
 
 static const char kInspectorName[] = "React Native V8 Inspector";
@@ -100,7 +98,7 @@ void InspectorFrontend::sendNotification(
   client_->SendRemoteMessage(message->string());
 }
 
-class LocalConnection : public react::ILocalConnection {
+class LocalConnection : public jsinspector::ILocalConnection {
  public:
   LocalConnection(std::weak_ptr<InspectorClient> weakClient)
       : weakClient_(weakClient) {}
@@ -193,11 +191,11 @@ void InspectorClient::ConnectToReactFrontend() {
   std::lock_guard<std::mutex> lock(connectionMutex_);
 
   auto weakClient = CreateWeakPtr();
-  pageId_ = react::getInspectorInstance().addPage(
+  pageId_ = jsinspector::getInspectorInstance().addPage(
       inspectorName_,
       "V8",
       [weakClient = std::move(weakClient)](
-          std::unique_ptr<react::IRemoteConnection> remoteConn) {
+          std::unique_ptr<jsinspector::IRemoteConnection> remoteConn) {
         auto client = weakClient.lock();
         if (client) {
           client->remoteConn_ = std::move(remoteConn);
@@ -212,7 +210,7 @@ void InspectorClient::Disconnect() {
   if (remoteConn_) {
     remoteConn_->onDisconnect();
   }
-  react::getInspectorInstance().removePage(pageId_);
+  jsinspector::getInspectorInstance().removePage(pageId_);
 }
 
 void InspectorClient::DisconnectFromReactFrontend() {
diff --git a/node_modules/react-native-v8/src/v8runtime/V8Inspector.h b/node_modules/react-native-v8/src/v8runtime/V8Inspector.h
index 7eeb2f1..15606f3 100644
--- a/node_modules/react-native-v8/src/v8runtime/V8Inspector.h
+++ b/node_modules/react-native-v8/src/v8runtime/V8Inspector.h
@@ -9,8 +9,14 @@
 
 #include <cxxreact/MessageQueueThread.h>
 #include <condition_variable>
-#include "jsinspector/InspectorInterfaces.h"
 #include "v8-inspector.h"
+#if REACT_NATIVE_TARGET_VERSION >= 73
+#include "jsinspector-modern/InspectorInterfaces.h"
+namespace jsinspector = facebook::react::jsinspector_modern;
+#else
+#include "jsinspector/InspectorInterfaces.h"
+namespace jsinspector = facebook::react;
+#endif
 
 namespace rnv8 {
 
@@ -84,7 +90,7 @@ class InspectorClient final
   v8::Isolate *isolate_;
   v8::Global<v8::Context> context_;
 
-  std::unique_ptr<facebook::react::IRemoteConnection> remoteConn_;
+  std::unique_ptr<jsinspector::IRemoteConnection> remoteConn_;
   std::string inspectorName_;
 
   std::mutex connectionMutex_;
```

**Review by Kudo — COMMENTED** [review]
i was thinking to pass ci, but given the ci is broken and don't have bandwidth to fix that. i would rather to build locally and it works fine.
there's just one remaining one recommended change, otherwise it's good to merge. thanks for helping the pr.

File: src/v8runtime/V8Inspector.h (hunk 4 above)
Comment (Serck96) [reply src/v8runtime/V8Inspector.h]: I agree with you, I've already made the change from "react" to "jsinspector".

**Review by Kudo — APPROVED** [review]
thanks again for the awesome work 🔥 let's ship it 🚀

**Comment by Kudo** [comment]
`react-native-v8@2.3.0` published!

**Pull request merged**